apps/api/src/persistence/repositories/CategoryRepository.e2e.spec.ts

```diff
@@ -0,0 +1,126 @@
+import { INestApplication } from '@nestjs/common';
+import createTestModule from '../../app/controllers/__tests__/createTestModule';
+import { InternalError } from '../../lib/errors/InternalError';
+import { NotFound } from '../../lib/types/not-found';
+import buildTestData from '../../test-data/buildTestData';
+import { ArangoConnectionProvider } from '../database/arango-connection.provider';
+import { DatabaseProvider } from '../database/database.provider';
+import { categoryCollectionID } from '../database/types/ArangoCollectionId';
+import { DatabaseDTO } from '../database/utilities/mapEntityDTOToDatabaseDTO';
+import generateRandomTestDatabaseName from './__tests__/generateRandomTestDatabaseName';
+import TestRepositoryProvider from './__tests__/TestRepositoryProvider';
+
+describe('Repository provider > getCategoryRepository', () => {
+    const testDatabaseName = generateRandomTestDatabaseName();
+
+    const testData = buildTestData();
+
+    let arangoConnectionProvider: ArangoConnectionProvider;
+
+    let databaseProvider: DatabaseProvider;
+
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let app: INestApplication;
+
+    const { categoryTree } = testData;
+
+    beforeAll(async () => {
+        jest.resetModules();
+
+        const moduleRef = await createTestModule(testDatabaseName);
+
+        arangoConnectionProvider =
+            moduleRef.get<ArangoConnectionProvider>(ArangoConnectionProvider);
+
+        databaseProvider = new DatabaseProvider(arangoConnectionProvider);
+
+        testRepositoryProvider = new TestRepositoryProvider(databaseProvider);
+
+        app = moduleRef.createNestApplication();
+
+        await app.init();
+    });
+
+    afterAll(async () => {
+        await arangoConnectionProvider.dropDatabaseIfExists();
+
+        await app.close();
+    });
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup;
+    });
+
+    afterEach(async () => {
+        await testRepositoryProvider.testTeardown();
+    });
+
+    describe('fetchTree', () => {
+        it('should return the expected result', async () => {
+            await testRepositoryProvider.addCategories(categoryTree);
+
+            const result = await testRepositoryProvider.getCategoryRepository().fetchTree();
+
+            expect(result).toEqual(categoryTree);
+        });
+
+        // TODO [test-coverage] Test that invalid documents lead to InternalErrors in result
```

Comment:
Since you are returning the errors, it is part of the API, and you should test it now.